content/docs/gke/pipelines/authentication-pipelines.md

```diff
@@ -5,53 +5,139 @@ weight = 50
 
-#### Cluster setup to use workload identity for Pipelines Standalone
+#### Cluster setup to use Workload Identity for Pipelines Standalone or AI Platform Pipelines
 
-**After you enabled workload identity**, you need to bind workload identities for KSAs used by [Pipelines Standalone](/docs/pipelines/installation/overview/#kubeflow-pipelines-standalone).
-The following helper bash scripts bind workload identities for KSAs provided by Pipelines Standalone:
+##### 1. Create your cluster with Workload Identity enabled
+
+* In Google Cloud Console UI, you can enable Workload Identity in `Create a Kubernetes cluster -> Security -> Enable Workload Identity` like the following:
+<img src="/docs/images/pipelines/gke-enable-workload-identity.png">
+
+* Using `gcloud` CLI, you can enable it with:
+```bash
+gcloud beta container clusters create cluster-name \
+  --release-channel regular \
+  --workload-pool=project-id.svc.id.goog
+```
+
+References:
+
+* [Enable Workload Identity on a new cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_workload_identity_on_a_new_cluster)
+
+* [Enable Workload Identity on an existing cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_workload_identity_on_an_existing_cluster)
+
+##### 2. Deploy Kubeflow Pipelines
+Deploy via [Pipelines Standalone](/docs/pipelines/installation/overview/#kubeflow-pipelines-standalone) or [AI Platform Pipelines](/docs/pipelines/installation/overview#gcp-hosted-ml-pipelines) as usual.
+
+Note, for AI Platform Pipelines additionally, you need to patch `proxy-agent` to work with workload identity:
+```
+kubectl patch deployment proxy-agent --patch '{"spec": {"template": {"spec": {"hostNetwork": true}}}}'
+```
+
+##### 3. Bind Workload Identities for KSAs used by Kubeflow Pipelines
+
+The following helper bash scripts bind Workload Identities for KSAs used by Kubeflow Pipelines:
 
 * [gcp-workload-identity-setup.sh](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/gcp-workload-identity-setup.sh) helps you create GSAs and bind them to KSAs used by pipelines workloads. This script provides an interactive command line dialog with explanation messages.
-* [wi-utils.sh](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/wi-utils.sh) provides minimal utility bash functions that let you customize your setup. The minimal utilities make it easy to read and use programmatically.
+* [wi-utils.sh](https://github.com/kubeflow/pipelines/blob/master/manifests/kustomize/wi-utils.sh) alternatively provides minimal utility bash functions that let you customize your setup. The minimal utilities make it easy to read and use programmatically.
+
+For example, to get a default setup using `gcp-workload-identity-setup.sh`, you can
+```
```

Comment:
KSA is created in Standalone Pipeline CLI's manifest, together with other deployment etc.
Right after kubectl apply, it will
(1) create KSAs
(2) launch deployments
(3) deployments can't be well launched if talks to GCP as KSA-GSA binding is not done yet.

Prefer change the sequence like these:
(1) Apply App CRD changes (same as before)
(2) Split the KSA from manifests. Create KSA first
(3) KSA-GSA binding.
(4) Other Deployments.

Aka bring the post-installation configure to installation phase to avoid (1) user forget; (2) errors before KSA-GSA binding which may scare away users.

Reply:
I think there's no need to separate, because

* KFP core system has 0 dependency on GCP (if not using managed storage)
* One of the benefit of Kubernetes's deployment model is self-healing, after bindings are set up, the system should become healthy soon by itself

maybe we can chat about this

Reply:
just cc'ed you an email about a concern on the failures. I'm OK if you strongly prefer keep them all together and leave it be self-healing until user do WI binding. Just that would let user worried about the installation status.

Reply:
We can address that with documentation, I think it's expected the system doesn't work until you configured all pieces together.